Nightly Dotty workflow of 2024-01-19 failed

See https://github.com/lampepfl/dotty/actions/runs/7578873365

```
 Test 'tests/run-staging/i19170b.scala' failed with output:                      
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at dotty.tools.vulpix.ChildJVMMain.runMain(ChildJVMMain.java:40)
	at dotty.tools.vulpix.ChildJVMMain.main(ChildJVMMain.java:47)
Caused by: java.lang.AssertionError: assertion failed: package scala.quoted.runtime.Expr does not have a member method quote
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at i19170b$package$.Test(i19170b.scala:16)
	at Test.main(i19170b.scala:10)
	... 6 more
```

This is the same issue that was patched for Windows in #19471

--- a/tests/run-staging/i19170b.scala
+++ b/tests/run-staging/i19170b.scala
@@ -8,7 +8,8 @@ class A(i: Int)
 def f(i: Expr[Int])(using Quotes): Expr[A] = { '{ new A($i) } }
 
 @main def Test = {
-  if !System.getProperty("os.name").contains("Windows") then
+  // The heuristic to give the extra information does not work on JDK 8
+  if System.getProperty("java.specification.version") != "1.8" then
     try
       val g: Int => A = staging.run { '{ (i: Int) => ${ f('{i}) } } }
       println(g(3))

